File: examples/blobs/bucket-manage.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List, inspect, and delete blobs in a bucket.
export async function manageBucket(client: JsBaoClient, blobId: string, expiredIds: string[]) {
  // #region example
  // List blobs in the bucket
  const { items, cursor } = await client.blobBuckets.list("avatars", { limit: 50 });

  // One blob's metadata
  const meta = await client.blobBuckets.getMetadata("avatars", blobId);

  // Delete a blob
  await client.blobBuckets.delete("avatars", blobId);

  // Delete a batch of blobs (up to 500 ids) in one call
  const batchResult = await client.blobBuckets.delete("avatars", expiredIds);
  // batchResult: { deleted, blobIds, bucketId }
  // #endregion example
  return { items, cursor, meta, batchResult };
}

File: examples/blobs/bucket-manage.swift
import JsBaoClient

// List, inspect, and delete blobs in a bucket.
func manageBucket(client: JsBaoClient, blobId: String, expiredIds: [String]) async throws {
  // #region example
  // List blobs in the bucket
  let page = try await client.blobBuckets.list(bucketIdOrKey: "avatars", limit: 50)
  let items = page.items
  let cursor = page.cursor

  // One blob's metadata
  let meta = try await client.blobBuckets.getMetadata(bucketIdOrKey: "avatars", blobId: blobId)

  // Delete a blob
  _ = try await client.blobBuckets.delete(bucketIdOrKey: "avatars", blobId: blobId)

  // Delete a batch of blobs (up to 500 ids) in one call
  let batchResult = try await client.blobBuckets.delete(bucketIdOrKey: "avatars", blobIds: expiredIds)
  // batchResult: BatchBlobDeleteResult { deleted, blobIds, bucketId }
  // #endregion example
  _ = (items, cursor, meta, batchResult)
}
